### PR TITLE
add integration test for user service

### LINKED ISSUE #28: Write integration test for user service

### PR DESCRIPTION
**Pull request by SachkovaYuliia**
Screenshot_1

### CODE REVIEW

File: src/test/integration/services/userService.spec.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        firstName: 'Eva',
+        lastName: 'Mendez',
+        email: 'em@gmailcom',
+        password: 'qwerty9876'
+      })
+      await user.save()
+
+      const retrievedUser = await userService.getUserById(user._id)
+      expect(retrievedUser).not.toBeNull()
+      expect(retrievedUser._id.toString()).toEqual(user._id.toString())
```
Comment (olehpona) [inline src/test/integration/services/userService.spec.js]: Is it necessary to check the string variant of the ObjectId here?
Maybe you can just use
```js
expect(retrievedUser._id).toEqual(user._id)
```

**Review by olehpona — APPROVED** [review]

**Review by grbvtsk — APPROVED** [review]

**Review by HalynaProlubshchykova — APPROVED** [review]

**Review by MrAK-dev — APPROVED** [review]

File: src/test/integration/services/userService.spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+const User = require('~/models/user')
+const { createError } = require('~/utils/errorsHelper')
+const { DOCUMENT_NOT_FOUND, ALREADY_REGISTERED } = require('~/consts/errors')
+
+let mongoServer
+
+beforeAll(async () => {
+  mongoServer = await MongoMemoryServer.create()
+  const uri = mongoServer.getUri()
+  await mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true })
```
Comment (vlad-khrychov) [inline src/test/integration/services/userService.spec.js]: I already saw the similar code. Create issue to move this logic to the some external file, as test setup.

File: src/test/integration/services/userService.spec.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      expect(retrievedUser).toBeNull()
+    })
+
+    it('should throw DOCUMENT_NOT_FOUND error if user not found', async () => {
+      const nonExistentId = mongoose.Types.ObjectId()
+      try {
+        await userService.getUserById(nonExistentId)
+      } catch (error) {
+        expect(error).toEqual(createError(404, DOCUMENT_NOT_FOUND([User.modelName])))
+      }
```
Comment (vlad-khrychov) [inline src/test/integration/services/userService.spec.js]: We discussed that it is not correct way to test errors.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
After fixing all comments do not forget write to the discord chat about it.

File: src/test/integration/setupTests.js
Comment (vlad-khrychov) [inline src/test/integration/setupTests.js:1]: Good approach, discuss with the team, maybe someone wants to modify previous tests with this setup. Or they should be aware of this possibility for the next tests.

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**